### PR TITLE
pod-scaler: deploy admission webhook

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

Part of [DPTP-1822](https://issues.redhat.com/browse/DPTP-1822)

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Seems to work (deployed the Pods, but not the webhook, on app.ci):

```
$ oc --context app.ci --namespace ci port-forward pod-scaler-consumer-admission-6bdcbb9bdc-z47dd :8080 &
Forwarding from 127.0.0.1:45497 -> 8080
Forwarding from [::1]:45497 -> 8080
Handling connection for 45497
$ curl -H "Content-Type: application/json" -X POST --data @/tmp/data 127.0.0.1:45497/pods
{"kind":"AdmissionReview","apiVersion":"admission.k8s.io/v1","response":{"uid":"705ab4f5-6393-11e8-b7cc-42010a800002","allowed":true,"status":{"metadata":{},"code":200}}}
$  oc --context app.ci --namespace ci logs pod-scaler-consumer-admission-6bdcbb9bdc-z47dd -f
time="2021-04-25T00:25:18Z" level=info msg="Initializing admission webhook server." component=admission
time="2021-04-25T00:25:18Z" level=info msg="Serving admission webhooks." component=admission
time="2021-04-25T00:38:22Z" level=debug msg="Handling labels on Pod created for a Build." build=manage-clonerefs-123 component=admission
```

**Review by petr-muller — APPROVED** [review]
/hold

LGTM with medium confidence :)

File: clusters/build-clusters/common/assets/pod-scaler-admission.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: "pod-scaling.ci.openshift.io"
+webhooks:
+- name: "pod-scaling.ci.openshift.io"
```
Comment (alvaroaleman) [inline clusters/build-clusters/common/assets/pod-scaler-admission.yaml:7]: yeah we should set objectSelector or namespaceSelector before we deploy this

Comment (stevekuznetsov) [reply clusters/build-clusters/common/assets/pod-scaler-admission.yaml:7]: I'm going to do it manually today on one cluster only

**Comment by stevekuznetsov** [comment]
```
I0427 04:21:43.532952       1 build_controller.go:1779] Giving up retrying skuznets/cli-jq: failed to create build pod: Internal error occurred: failed calling webhook "pod-scaling.ci.openshift.io": Post "https://pod-scaler-consumer-admission.ci.svc:443/?timeout=5s": x509: certificate signed by unknown authority
```

sad

File: clusters/build-clusters/common/assets/pod-scaler-admission.yaml [hunk 2]
```diff
@@ -0,0 +1,118 @@
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: "pod-scaling.ci.openshift.io"
+  annotations:
+    service.beta.openshift.io/inject-cabundle: "true"
+webhooks:
+- name: "pod-scaling.ci.openshift.io"
+  namespaceSelector:
```
Comment (alvaroaleman) [inline clusters/build-clusters/common/assets/pod-scaler-admission.yaml:10]: Can we set `failurePolicy: Ignore` to not make this a SPOF?

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**